src/index.ts

```diff
@@ -1,47 +1,54 @@
 import ejs from "ejs";
-import { getContext, helpers, Property } from "@dylibso/xtp-bindgen";
+import { helpers, getContext, ObjectType, EnumType, ArrayType, XtpNormalizedType, MapType, Parameter, Property } from "@dylibso/xtp-bindgen"
 
-function toGolangType(property: Property): string {
-  if (property.$ref) return goName(property.$ref.name);
-  switch (property.type) {
-    case "string":
-      if (property.format === "date-time") {
-        return "time.Time";
-      }
-      return "string";
-    case "number":
-      if (property.format === "float") {
-        return "float32";
-      }
-      if (property.format === "double") {
-        return "float64";
+function toGolangTypeX(type: XtpNormalizedType, required?: boolean): string {
+  // annotate with null if nullable 
+  const pointerify = (t: string) => {
+    // TODO key off the `type.kind` here instead
+    if (t.startsWith("[]") || t.startsWith("map[")) return t
```

Comment:
I borrowed this from the old pointerToGolangType function but will change it to kee off the type

Reply:
actually, this is kind of fine and closer represents the logic it means to convey. keying off the type
would work but be indirect and possibly get out of date.